File: src/client/types/invoice.ts
export type InvoiceStatus = "draft" | "sent" | "paid" | "overdue" | "cancelled";

export interface Invoice {
  id: string;
  invoiceNumber: string;
  companyId: string;

  // Client details
  clientId?: string;
  clientName: string;
  clientEmail: string;
  clientAddress?: string;

  // Invoice details
  issueDate: string;
  dueDate: string;
  subTotal: number;
  taxRate: number;
  taxAmount: number;
  totalAmount: number;
  status: InvoiceStatus;
  notes?: string;

  // Payment details
  paymentTerms?: string;
  paymentInstructions?: string;

  // Audit
  createdAt: string;
  updatedAt: string;

  // Items
  items: InvoiceItem[];
}

export interface InvoiceItem {
  id: string;
  invoiceId: string;
  description: string;
  quantity: number;
  unitPrice: number;
  amount: number;
}

export interface InvoiceTableState {
  currentPage: number;
  itemsPerPage: number;
  totalPages: number;
  totalItems: number;
}

export interface InvoiceFilters {
  clientName?: string;
  status?: InvoiceStatus;
  dateRange?:
    | {
        start?: string;
        end?: string;
      }
    | undefined;
  keyword?: string;
  searchField?: "all" | "number" | "description";
}

File: src/api/Models/Invoice.cs
namespace InvoiceGeneratorAPI.Models;

public class Invoice
{
    public Guid Id { get; set; }
    public string InvoiceNumber { get; set; } = string.Empty;

    // Company relationship
    public Guid CompanyId { get; set; }
    public Company Company { get; set; } = null!;

    // Client relationship (optional, can use client details directly)
    public Guid? ClientId { get; set; }
    public Client? Client { get; set; }

    // Client details (used when no stored client or for historical record)
    public string ClientName { get; set; } = string.Empty;
    public string ClientEmail { get; set; } = string.Empty;
    public string? ClientAddress { get; set; }

    // Invoice details
    public DateTime IssueDate { get; set; }
    public DateTime DueDate { get; set; }
    public decimal SubTotal { get; set; }
    public decimal TaxRate { get; set; }
    public decimal TaxAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = "Draft"; // Draft, Sent, Paid, Overdue, Cancelled
    public string? Notes { get; set; }

    // Payment details
    public string? PaymentTerms { get; set; }
    public string? PaymentInstructions { get; set; }

    // Audit
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Items
    public List<InvoiceItem> Items { get; set; } = new();
}

public class InvoiceItem
{
    public Guid Id { get; set; }
    public Guid InvoiceId { get; set; }
    public Invoice Invoice { get; set; } = null!;
    public string Description { get; set; } = string.Empty;
    public decimal Quantity { get; set; }
    public decimal UnitPrice { get; set; }
    public decimal Amount { get; set; }
}